Source/WebCore/html/HTMLElement.cpp

```diff
@@ -1449,6 +1464,9 @@ ExceptionOr<void> HTMLElement::hidePopoverInternal(FocusPreviousElement focusPre
 
     removeFromTopLayer();
 
+    if (popoverState() == PopoverState::Auto)
+        document().autoPopoverList().removeLast();
```

Comment:
If the element being removed from the top layer isn't the last one, this is wrong.